sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -700,6 +704,8 @@ def fit(self, X, y, sample_weight=None):
                     max_depth=self.max_depth,
                     min_samples_leaf=self.min_samples_leaf,
                     l2_regularization=self.l2_regularization,
+                    feature_fraction_per_split=self.max_features,
+                    rng=np.random.default_rng(self._random_seed2),
```

Comment:
I'm not too familiar with Generator objects, but doesn't that mean that each Grower will have the exact same RNG for feature subsampling? Is this intended?

Reply:
Good catch, indeed, I would have expected that the rng instance would have been created before entering the `for iteration in range(begin_at_stage, self.max_iter):` loop at line 651.

Reply:
Well spotted. I put the definition directly after defining `self._random_seed2`.